crates/bevy_math/src/rotation2d.rs

```diff
@@ -0,0 +1,332 @@
+use glam::FloatExt;
+
+use crate::prelude::{Direction2d, Mat2, Vec2};
+
+/// A counterclockwise 2D rotation in radians.
+///
+/// The rotation angle is wrapped to be within the `]-pi, pi]` range.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+pub struct Rotation2d {
+    /// The cosine of the rotation angle in radians.
+    ///
+    /// This is the real part of the unit complex number representing the rotation.
+    pub cos: f32,
+    /// The sine of the rotation angle in radians.
+    ///
+    /// This is the imaginary part of the unit complex number representing the rotation.
+    pub sin: f32,
+}
+
+impl Default for Rotation2d {
+    fn default() -> Self {
+        Self::IDENTITY
+    }
+}
+
+impl Rotation2d {
+    /// No rotation.
+    pub const IDENTITY: Self = Self { cos: 1.0, sin: 0.0 };
+
+    /// A rotation of π radians.
+    pub const PI: Self = Self {
+        cos: -1.0,
+        sin: 0.0,
+    };
+
+    /// A counterclockwise rotation of π/2 radians.
+    pub const FRAC_PI_2: Self = Self { cos: 0.0, sin: 1.0 };
+
+    /// A counterclockwise rotation of π/3 radians.
+    pub const FRAC_PI_3: Self = Self {
+        cos: 0.5,
+        sin: 0.866_025_4,
+    };
+
+    /// A counterclockwise rotation of π/4 radians.
+    pub const FRAC_PI_4: Self = Self {
+        cos: std::f32::consts::FRAC_1_SQRT_2,
+        sin: std::f32::consts::FRAC_1_SQRT_2,
+    };
+
+    /// A counterclockwise rotation of π/6 radians.
+    pub const FRAC_PI_6: Self = Self {
+        cos: 0.866_025_4,
+        sin: 0.5,
+    };
+
+    /// A counterclockwise rotation of π/8 radians.
+    pub const FRAC_PI_8: Self = Self {
+        cos: 0.923_879_5,
+        sin: 0.382_683_43,
+    };
+
+    /// Creates a [`Rotation2d`] from a counterclockwise angle in radians.
+    ///
+    /// This is equivalent to [`Self::from_radians`].
+    #[inline]
+    pub fn new(radians: f32) -> Self {
+        Self::from_radians(radians)
+    }
```

Comment:
I removed it for now. We could also remove the `from_` to make the constructors shorter:

```rust
// Before
let rotation1 = Rotation2d::from_radians(PI / 2.0);
let rotation2 = Rotation2d::from_degrees(45.0);

// After
let rotation1 = Rotation2d::radians(PI / 2.0);
let rotation2 = Rotation2d::degrees(45.0);
```

The getters would still be `as_radians` and `as_degrees`. No strong preference either way for me though

Reply:
I changed it to the shorter one for now, I think it might look a bit nicer. It should be pretty obvious that it's a constructor and not a getter considering that it takes an argument and there's already `as_radians` and `as_degrees`.